Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionCiscoASA.yml

```diff
@@ -0,0 +1,369 @@
+          , "255","Reserved"];
+  let NWParser=(disabled:bool=false){
+      let allLogs = CommonSecurityLog
+      | where not(disabled)
+      | where DeviceVendor == "Cisco" and DeviceProduct == "ASA"
+      | project-away Activity, AdditionalExtensions, ApplicationProtocol, DeviceFacility, DeviceAddress, DeviceName,DestinationDnsDomain, DestinationServiceName, DestinationTranslatedAddress, DestinationTranslatedPort, DeviceDnsDomain, DeviceExternalID, DeviceInboundInterface, DeviceNtDomain, DeviceOutboundInterface, DevicePayloadId, DeviceTranslatedAddress, DestinationHostName, DestinationMACAddress, DestinationNTDomain, DestinationProcessId, DestinationProcessName, DestinationUserID, DestinationUserName, DestinationUserPrivileges, DeviceMacAddress, EndTime, ExternalID, File*, Flex*, IndicatorThreatType, Malicious*, Old*, Process*, ReceiptTime, ReceivedBytes, Remote*, Report*, Request*, SourceServiceName, SourceTranslatedAddress, SourceTranslatedPort, SourceUserID, SourceUserName, SourceUserPrivileges, SourceDnsDomain, SourceHostName, SourceMACAddress, SourceProcessId, SourceProcessName, SourceSystem, SentBytes, StartTime, TenantId, ThreatConfidence, ThreatDescription, ThreatSeverity, DeviceCustomDate*, DeviceCustomFloatingPoint*, DeviceCustomIPv6Address*, DeviceCustomNumber*, DeviceTimeZone;
+      let all_106001_alike = allLogs
+                | where DeviceEventClassID in ("106001", "106006", "106015", "106016", "106021", "106022") 
+                | parse Message with * " interface " DstInterfaceName;
+      let all_106010_alike = allLogs
+                | where DeviceEventClassID in ("106010", "106014")
+                | parse Message with * " src "  SrcInterfaceName ":" * " dst " DstInterfaceName ":" * "(type " NetworkIcmpType ", code " NetworkIcmpCode:int ")";
+      let all_106018 = allLogs
+                | where DeviceEventClassID == "106018"
+                | parse Message with * " packet type " NetworkIcmpType " " * "list " NetworkRuleName " " *;
+      let all_106023 = allLogs
+                | where DeviceEventClassID == "106023" and not(Message has "protocol 41") and isnotempty(SourceIP)
+                | parse Message with * " src "  SrcInterfaceName ":" * " dst " DstInterfaceName ":" * ' by access-group "' NetworkRuleName '" '  *
+                | parse Message with * "(type " NetworkIcmpType ", code " NetworkIcmpCode:int ")" *;
+      let all_106023_unparsed = allLogs
+                | where DeviceEventClassID == "106023" and isempty(SourceIP) and not(Message has "protocol 41")
+                | parse Message with * ": " DeviceAction " " Protocol " src " SrcInterfaceName ":" SrcIpAddrAndPort "(" SrcUsername ") dst " DstInterfaceName ":" DstIpAddrAndPort " " NetworkIcmpInfo 'by access-group "' NetworkRuleName '" [' * "]"
+                | parse NetworkIcmpInfo with "(type " NetworkIcmpType ", code " NetworkIcmpCode:int ") "
+                | extend SrcIpAddr = tostring(split(SrcIpAddrAndPort,"/",0)[0]),
```

Comment:
Not critical, but using a temp dynamic for the split, so don' split the same fields twice is more efficient.

Reply:
Updated